### DISCUSSION TITLE
Unable to use binary in GitHub workflow

### DISCUSSION THREAD

**Discussion by armandgrillet** [community/community#25392] [Actions]
<p>Hi,</p>
<p>I’m trying to use a binary available in my container in a GitHub workflow. My yaml looks like this:</p>
<pre><code class="lang-auto">jobs:
  speed:
    name: Speed
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: ${{ github.workspace }}
    container: armandgrillet/platinum-speed:latest
    steps:
    - uses: actions/checkout@v1
    - name: Try which
      run: which ack
    - name: Use ack
      run: ack test
</code></pre>
<p>The container, <code>armandgrillet/platinum-speed:latest</code>, is based on Ubuntu and has <code>ack</code> installed using <code>apt-get install</code>.</p>
<p>When running the workflow described above, the step <code>Try which</code> returns <code>/usr/bin/ack</code> as expected. However, when running the second step (which I tested by running a container using the same image as the one in the workflow), I get:</p>
<pre><code class="lang-auto">Run ack test
  ack test
  shell: sh -e {0}
##[error]Process completed with exit code 1.
</code></pre>
<p>What am I missing?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/armandgrillet">@armandgrillet</a>,</p>
<p>The error is due to command ‘ack’ doesn’t exit properly.<br>
Please add ‘&amp;exit 0’ to make sure it exits properly. Code sample as below:</p>
<pre><code class="lang-auto">    - name: Use ack
      run: |
        ack test &amp; exit 0
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181090738-e6671414-e40d-47e4-8541-6a983fb1e431.png" alt="ackcommand" data-base62-sha1="6LUy2eWpUfmM7p4ZDMiuaKhKbr6" width="655" height="446"></p>
<p>Thanks</p>
